lib/solarwinds_apm/opentelemetry/solarwinds_propagator.rb

```diff
@@ -80,12 +83,10 @@ def inject(carrier, context: ::OpenTelemetry::Context.current,
             end
             setter.set(carrier, TRACESTATE_HEADER_NAME, Utils.trace_state_header(trace_state))
           end
+        rescue StandardError => e
+          SolarWindsAPM.logger.debug { "[#{self.class}/#{__method__}] Injection failed: #{e.message}" }
         end
 
-        # Returns the predefined propagation fields. If your carrier is reused, you
-        # should delete the fields returned by this method before calling +inject+.
-        #
-        # @return [Array<String>] a list of fields that will be used by this propagator.
```

Comment:
Thanks for adding back! Would be good to keep this comment too, maybe with a change so we remember next time

```
        # Returns the predefined propagation fields, required by upstream.
        # If your carrier is reused, you should delete the fields returned by
        # this method before calling +inject+.
        #
        # @return [Array<String>] a list of fields that will be used by this propagator.
```